actions/promote/promote.js

```diff
@@ -17,28 +17,51 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies
 const openwhisk = require('openwhisk');
-const { getAioLogger } = require('../utils');
+const { getAioLogger, updateStatusToStateLib, PROMOTE_ACTION } = require('../utils');
 
 // This returns the activation ID of the action that it called
 function main(args) {
     const logger = getAioLogger();
     let payload;
+    const {
+        spToken, adminPageUri, projectExcelPath, projectRoot
+    } = args;
     try {
-        const { spToken, adminPageUri, projectExcelPath } = args;
-        if (!spToken || !adminPageUri || !projectExcelPath) {
+        if (!projectRoot) {
             payload = 'Required data is not available to proceed with FG Promote action.';
             logger.error(payload);
+        } else if (!spToken || !adminPageUri || !projectExcelPath) {
+            payload = 'Required data is not available to proceed with FG Promote action.';
+            updateStatusToStateLib(projectRoot, 'Failure', payload, '', PROMOTE_ACTION);
+            logger.error(payload);
         } else {
+            updateStatusToStateLib(projectRoot, 'In-Progress', 'Triggering promote action', '', PROMOTE_ACTION);
             const ow = openwhisk();
             return ow.actions.invoke({
                 name: 'milo-fg/promote-worker',
                 blocking: false, // this is the flag that instructs to execute the worker asynchronous
                 result: false,
                 params: args
+            }).then((result) => {
+                logger.info(result);
+                //  attaching activation id to the status
+                updateStatusToStateLib(projectRoot, 'In-Progress', '', result.activationId, PROMOTE_ACTION);
+                return {
+                    code: 200,
+                    body: { Success: result },
+                };
+            }).catch((err) => {
+                updateStatusToStateLib(projectRoot, 'Failure', `Failed to invoke actions ${err.message}`, '', PROMOTE_ACTION);
+                logger.error('failed to invoke actions', err);
+                return {
+                    code: 500,
+                    body: { Error: err }
+                };
             });
         }
     } catch (err) {
         logger.error(err);
+        updateStatusToStateLib(projectRoot, 'Failure', `Failed to invoke actions ${err.message}`, '', PROMOTE_ACTION);
```

Comment:
As discussed, create a const for the Project Status values similar to the one we have here:
https://github.com/adobecom/milo/blob/main/tools/floodgate/js/project.js#L16-L20

Also create const's for any strings that are getting re-used for logging or otherwise.